src/ahp/prover.rs

```diff
@@ -262,23 +204,32 @@ impl<F: PrimeField> AHPForR1CS<F> {
         ]);
 
         let w_poly_time = start_timer!(|| "Computing w polynomial");
+
+        // Compute the domain evaluations of w(X) - x(X) on H \ I by from
+        // the witness vector w and x, and set it zero on the input domain I.
+        // TODO: Let us use reindex_by_subdomain.
```

Comment:
What was the reason why we don't use reindex by subdomain?

Reply:
I think that the mapping provided by `reindex_by_subdomain` cannot be directly used in the indexing of `w_extended`. Instead we would have to introduce a new function, let's call it `reindex_by_subdomain_inv`, which implements the inverse mapping. In that case the else branch would become
```
else {
    // The mapping of the witness vector to values on H.
    let w_index = domain_h.reindex_by_subdomain_inv(domain_x.size(), k);
    w_extended[w_index] - &x_evals[k]
}
```
Shall we do it?